テーブル設計

AirLabCloud opened the pull request:
README.mdへのマークダウンの記述が完了しました。
お目通しをお願いします。
以下がERDの画像URLです。
https://gyazo.com/62aadd7a82133afc1bc290de6031c270

AirLabCloud committed:
テーブル設計

isyt729 commented on README.md:
```diff
@@ -1,24 +1,75 @@
+| :-------------------: | :----: | :-----------------------: |
+| nickname              | string | null: false               |
+| email                 | string | null: false, unique: true |
+| password              | string | null: false, unique: true |
+| password_confirmation | string | null: false, unique: true |
```
passwordはdeviseにてencrypted_passwordカラムが自動生成されるので、設計も合わせておきましょう！

Akinori0123 replied:
前回コードレビューで命名規則通りに記述を合わせていただきありがとうございます。
命名規則は今後も意識していただければと思いますが、deviseではpasswordやpassword_confirmationは不要となりますので削除していただいて問題ありません。
過去のアプリ（pictweet等）と同じくencrypted_passwordカラムを使う形になりますのでencrypted_passwordカラムを記述いただければと思います。

isyt729 commented on README.md:
```diff
@@ -1,24 +1,75 @@
+| Column           | Type       | Options                       |
+| :--------------: | :--------: | :---------------------------: |
+| user             | references | null: false, foreign_key: true|
+| images_id        | integer    | null: false                   |
```
| images_id        | integer    | null: false                   |

画像はactive_storageで管理しますので、このカラムは不要です！

isyt729 commented on README.md:
```diff
@@ -1,24 +1,75 @@
+- belongs_to :product
+- belongs_to_active_hash :prefecture
+
+## images テーブル
+| Column  | Type       | Options                        |
+| :-----: | :--------: | :----------------------------: |
+| product | references | null: false, foreign_key: true |
+| image   | string     | null: false                    |
```
画像はactive_storageで管理しますので、imagesテーブルは不要です！

isyt729 commented on README.md:
```diff
@@ -1,24 +1,75 @@
+- belongs_to :user
+- belongs_to :product
+- belongs_to_active_hash :prefecture
+
```
購入情報を管理するテーブルを追加しましょう！

isyt729 reviewed (commented):
レビュー依頼ありがとうございます！
コメントしましたので、ご確認お願いします！

AirLabCloud committed:
テーブル設計の修正

1.usersテーブルのpasswordカラムを削除。
2.imagesテーブルを削除し、ordersテーブルを追加。
3.付随するアソシエーションを削除及び追加。
4.各テーブルのforeign_keyの記述をFKと短縮。

Syota-Yamaguchi commented:
該当箇所を修正しましたので、再度お目通しをお願いします。

Katsuno0814 commented on README.md:
```diff
@@ -21,24 +19,22 @@
+## Association
 
 - belongs_to :user
 - has_one : purchasers
```
has_one : purchaserとhas_oneの場合は単数系で書きましょう！

Katsuno0814 commented on README.md:
```diff
@@ -47,29 +43,32 @@
 
 ### Association
 
 - belongs_to :user
```
- belongs_to :user
belongs_to :product
こちらは削除いたしましょう！

Katsuno0814 commented on README.md:
```diff
@@ -47,29 +43,32 @@
+| product | references | null: false, FK: true |
 
 ### Association
 
```
has_one : purchaser
を追加いたしましょう！

Katsuno0814 commented on README.md:
```diff
@@ -47,29 +43,32 @@
 - belongs_to :user
 - belongs_to :product
 - belongs_to_active_hash :prefecture
+- has_one :order
```
belongs_to :order
を追加しましょう！
またorderの外部キーカラムの追加もお願いいたします！

Katsuno0814 reviewed (commented):
レビュー依頼ありがとうございます！
コメントしましたので、ご確認をお願いいたします！

AirLabCloud committed:
テーブル設計の再修正

Syota-Yamaguchi commented:
指摘箇所の修正完了しました。
再度お目通しをお願い致します。

chirikuzu commented on README.md:
```diff
@@ -1,24 +1,76 @@
-* Deployment instructions
+| Column        | Type       | Options                   |
+| :-----------: | :--------: | :-----------------------: |
+| product       | references | null: false, FK: true     |
+| user          | references | null: false, FK: true     |
```
今回の設計では「配送先住所と購入管理が紐づく」実装を行なっていきたいので、商品やユーザーの外部キーは配送先住所テーブルには必要ないかと考えられます。

chirikuzu reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました。
追加でコメントいたしましたので、ご確認お願いします！

AirLabCloud committed:
テーブル設計の再修正

Syota-Yamaguchi commented:
指摘箇所の修正を行いました。
お目通しをお願い致します。

i-date commented on README.md:
```diff
@@ -1,24 +1,74 @@
-application up and running.
+## users テーブル
 
-Things you may want to cover:
+| Column                | Type   | Options                   |
+| :-------------------: | :----: | :-----------------------: |
+| nickname              | string | null: false               |
+| email                 | string | null: false, unique: true |
+| family_name           | string | null: false               |
+| first_name            | string | null: false               |
+| family_name_kana      | string | null: false               |
+| first_name_kana       | string | null: false               |
+| birth_day             | date   | null: false               |
```
usersテーブルにpasswordカラムが存在しないため、passwordカラムをusersテーブルに追加しましょう。

i-date commented on README.md:
```diff
@@ -1,24 +1,74 @@
 
-* System dependencies
+- has_many :products
+- has_many :purchasers
```
今回の設計では「1つの購入情報は１つの配送先住所を持っている」という設計、つまり、purchasersテーブルはordersテーブルとだけアソシエーションを持つという関係にしたいです。

そのため、usersテーブルの`has_many :purchasers`は削除しましょう。
そのかわり、購入情報を扱っているordersテーブルとのアソシエーション`has_many :orders`を、usersテーブルに追加しましょう。

レビュアー側でpurchasersテーブルと、ordersテーブルを混同してしまい失礼いたしました。

i-date commented on README.md:
```diff
@@ -1,24 +1,74 @@
 
-* How to run the test suite
+- belongs_to :user
+- has_one : purchaser
```
同様に今回の設計では「1つの購入情報は１つの配送先住所を持っている」という設計、つまり、purchasersテーブルはordersテーブルとだけアソシエーションを持つという関係にしたいです。

そのため、itemsテーブルの`has_one : purchaser`は削除しましょう。
そのかわり、購入情報を扱っているordersテーブルとのアソシエーション`has_one :order`を、itemsテーブルに追加しましょう。

レビュアー側でpurchasersテーブルと、ordersテーブルを混同してしまい失礼いたしました。

i-date reviewed (commented):
修正ありがとうございます！
少し追加コメントいたしましたので、ご確認お願いいたします。

AirLabCloud committed:
テーブル設計の再修正

Syota-Yamaguchi commented:
指摘箇所の修正が完了しました。
お目通しをお願い致します。

i-date commented on README.md:
```diff
@@ -6,6 +6,8 @@
 | nickname              | string | null: false               |
 | email                 | string | null: false, unique: true |
+| password              | string | null: false, unique: true |
+| password confirmation | string | null: false, unique: true |
```
10行目のカラム名ですが、複数単語になる場合はrailsの命名規則に基づき、`_`でつなげるようにしましょう。

i-date reviewed (commented):
修正ありがとうございます！
一点コメントいたしましたので、ご確認お願いいたします。

AirLabCloud committed:
テーブル設計の再修正

Syota-Yamaguchi commented:
指摘箇所の修正が完了しました。
再度お目通しをお願い致します。

Akinori0123 commented on app/models/category.rb:
```diff
@@ -0,0 +1,2 @@
+class Category < ApplicationRecord
+end
```
**app/models/内に作成されたモデル全てなど、readme以外の変更**
今回はテーブル設計のプルリクエストとなるため、readme以外の変更は不要となります。
もし残さないといけない理由がなければactive_hashの導入などは一度削除していただいた方が良いかと考えられます。
（今active_hashを導入してしまうと後々active_hashの部分をレビューできなくなってしまうため）

以下のプルリクエストでのfile変更内容がなるべくreadmeのみになるようにしましょう。
https://github.com/Syota-Yamaguchi/furima-29271/pull/1/files

Akinori0123 reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました。追加でコメントさせていただきましたので、ご確認いただけますと幸いです（同じ箇所の指摘を繰り返してしまい申し訳ありません）
よろしくお願いいたします。

AirLabCloud committed:
テーブル設計の再修正

1.usersテーブルからpasswordカラムを削除し、encrypted_passwordカラムを追加しました。
2.modelsフォルダから、active_hashのrbフォルダを削除しました。

Syota-Yamaguchi commented:
指摘箇所の修正が完了しました。
宜しくお願いします。

chirikuzu commented:
レビュー依頼ありがとうございます！
LGTMとさせていただきます！実装お疲れ様でした！
引き続き実装頑張っていきましょう！
